lib/pbench/server/api/resources/query_apis/metadata_index/timeseries_samples.py

```diff
@@ -0,0 +1,232 @@
+from http import HTTPStatus
+from logging import Logger
+from typing import AnyStr
+
+from flask import Response, jsonify
+from flask_restful import abort
+
+from pbench.server import PbenchServerConfig
+from pbench.server.api.resources import (
+    JSON,
+    Parameter,
+    ParamType,
+    Schema,
+)
+from pbench.server.api.resources.query_apis import CONTEXT, PostprocessError
+from pbench.server.api.resources.query_apis.metadata_index import RunIdBase
+from pbench.server.database.models.template import Template, TemplateNotFound
+
+
+class TimeSeriesSamples(RunIdBase):
```

Comment:
Without comparing line-by-line ... it doesn't appear this is much different from the sample data query, except with a different index and document time. Given that the filters are constructed from a namespace query (or from out-of-band knowledge of the schema), could these be consolidated with, for example, a base class `__init__` that takes the root index name to use?